rfc-parser/src/main/java/tech/pantheon/yanginator/parser/services/GrammarKitRFCUtils.java

```diff
@@ -0,0 +1,342 @@
+        }
+
+        final Pattern asteriskRegex = Pattern.compile(".*(?<wordSelect>" + regexInsert + ").*");
+        List<String> result = new ArrayList<>();
+        for (String line : lines) {
+            Matcher matcher = asteriskRegex.matcher(line.trim());
+            while (matcher.matches()) {
+                String oldWord = matcher.group("wordSelect");
+                String newWord = oldWord.replace(oldOperator, "");
+                newWord = newWord.concat(replacement);
+                line = line.replace(oldWord, newWord);
+                matcher = asteriskRegex.matcher(line.trim());
+            }
+            result.add(line);
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a list of string in which string rules are modified. Each string rule
+     * in provided list contains commentaries enclosed in angle brackets ("<...>").
+     * These commentaries are replaced by actual rules.
+     *
+     * @param lines the list of strings in which string rules are to be replaced.
+     * @return the list of strings with string rules replaced.
+     */
+    public static List<String> rewriteStringRules(final List<String> lines) {
+        final Pattern rulePattern = Pattern.compile(".*(?<statementStart>(<[\\s+\\w].*))$");
+        List<String> result = new ArrayList<>();
+        for (int i = 0; i < lines.size(); i++) {
+            String firstLineOfRule = lines.get(i);
+            Matcher matcher = rulePattern.matcher(firstLineOfRule.trim());
+            if (matcher.matches()) {
+                String originalString = matcher.group("statementStart");
+                String quotes = "\"\\\"\"";
+                String secondLineOfRule = lines.get(i + 1);
+                if (secondLineOfRule.contains("-arg")) {
+                    secondLineOfRule = secondLineOfRule.trim().replace(">", quotes);
+                    String replacement = "";
+                    firstLineOfRule = firstLineOfRule.replace(originalString, replacement.concat(quotes).concat(" " + secondLineOfRule));
+                } else {
+                    firstLineOfRule = firstLineOfRule
+                            .replace("<", "//")
+                            .concat(" " + secondLineOfRule.trim())
+                            .replace(">", "");
+                }
+                i++;
+            }
+            result.add(firstLineOfRule);
+        }
+
+        return result;
+    }
+
+    /**
+     * Replaces each occurrence of a hexadecimal range of values with corresponding characters.
+     *
+     * @param lines the list of strings in which range of hexadecimal values will be replaced.
+     * @return the list of strings in which range of hexadecimal values are replaced by characters.
+     */
+    public static List<String> replaceHexadecimalRange(List<String> lines) {
+        List<String> result = new ArrayList<>();
+        final Pattern hexPatternRange = Pattern.compile("(?<hexadecimalRange>(%x\\d{1,2}\\w-\\d{1,2}\\w))");
```

Comment:
Patterns should be always constants so they are only compiled once.

Reply:
This means, I should create constant, like this example, outside of this method for each pattern in this class?

`private static final Pattern HEXADECIMAL_RANGE_PATTERN = Pattern.compile("(?<hexadecimalRange>(%x\\d{1,2}\\w-\\d{1,2}\\w))");`

Reply:
yes, please.

Reply:
Fixed.